src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -319,6 +319,33 @@ void Parse::dump_map_adr_mem() const {
 
 #endif
 
+#include "ci/ciUtilities.inline.hpp"
+#include "compiler/methodMatcher.hpp"
+
+class PEAContext {
+private:
+  BasicMatcher* _matcher;
+
+  PEAContext() {
+    if (PEAMethodOnly != nullptr) {
```

Comment:
for these null checks can we just use 

```
if (PEAMethodOnly) {
```

Reply:
https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md#miscellaneous